src/PrpShop/QPlasmaTreeItem.cpp

```diff
@@ -15,88 +15,177 @@
-      fHasTextures(false), fAge(age)
+      fHasTextures(false), fAge(age), fAgeSeqPrefix(ageSeqPrefix)
 {
-    setText(0, age);
-    setIcon(0, QIcon(":/img/age.png"));
+    reinit();
 }
 
 QPlasmaTreeItem::QPlasmaTreeItem(QTreeWidget* parent, plPageInfo* page)
     : QTreeWidgetItem(parent, kTypePage), fPage(page)
 {
-    setText(0, st2qstr(page->getPage()));
-    setIcon(0, QIcon(":/img/page.png"));
+    reinit();
 }
 
 QPlasmaTreeItem::QPlasmaTreeItem(QTreeWidgetItem* parent)
     : QTreeWidgetItem(parent, kTypeNone)
 {
-    setIcon(0, QIcon(":/img/folder.png"));
+    reinit();
 }
 
 QPlasmaTreeItem::QPlasmaTreeItem(QTreeWidgetItem* parent, plKey obj)
-    : QTreeWidgetItem(parent, kTypeKO), fObjKey(obj)
+    : QTreeWidgetItem(parent, kTypeKO), fObjKey(std::move(obj))
 {
-    setText(0, st2qstr(obj->getName()));
-    setIcon(0, pqGetTypeIcon(obj->getType()));
+    reinit();
 }
 
-QPlasmaTreeItem::QPlasmaTreeItem(QTreeWidgetItem* parent, const QString& age)
+QPlasmaTreeItem::QPlasmaTreeItem(QTreeWidgetItem* parent, short classType)
+    : QTreeWidgetItem(parent, kTypeClassType), fClassType(classType)
+{
+    reinit();
+}
+
+QPlasmaTreeItem::QPlasmaTreeItem(QTreeWidgetItem* parent, const QString& age, int ageSeqPrefix)
     : QTreeWidgetItem(parent, kTypeAge), fHasBuiltIn(false),
-      fHasTextures(false), fAge(age)
+      fHasTextures(false), fAge(age), fAgeSeqPrefix(ageSeqPrefix)
 {
-    setText(0, age);
-    setIcon(0, QIcon(":/img/age.png"));
+    reinit();
 }
 
 QPlasmaTreeItem::QPlasmaTreeItem(QTreeWidgetItem* parent, plPageInfo* page)
     : QTreeWidgetItem(parent, kTypePage), fPage(page)
 {
-    setText(0, st2qstr(page->getPage()));
-    setIcon(0, QIcon(":/img/page.png"));
+    reinit();
+}
+
+static QString pqFormatPageName(const plPageInfo* page)
+{
+    if (s_showAgePageIDs) {
+        return st2qstr(ST::format("{} {}", page->getLocation().toString(), page->getPage()));
+    } else {
+        return st2qstr(page->getPage());
+    }
+}
+
+static QString pqFormatAgeName(const QString& name, int seqPrefix)
+{
+    if (s_showAgePageIDs) {
+        return QString("<%1> %2").arg(seqPrefix).arg(name);
+    } else {
+        return name;
+    }
+}
+
+void QPlasmaTreeItem::reinit()
+{
+    switch (type()) {
+        case kTypeNone:
+            setIcon(0, QIcon(":/img/folder.png"));
+            break;
+
+        case kTypeKO:
+            setText(0, st2qstr(fObjKey->getName()));
+            setIcon(0, pqGetTypeIcon(fObjKey->getType()));
+            break;
+
+        case kTypeClassType:
+            setText(0, pqGetFriendlyClassName(fClassType));
+            setIcon(0, QIcon(":/img/folder.png"));
+            break;
+
+        case kTypeAge:
+            setText(0, pqFormatAgeName(fAge, fAgeSeqPrefix));
+            setIcon(0, QIcon(":/img/age.png"));
+            break;
+
+        case kTypePage:
+            setText(0, pqFormatPageName(fPage));
+            setIcon(0, QIcon(":/img/page.png"));
+            break;
```

Comment:
Should probably add a default case with an assert to catch if/when the enum is modified in the future.